app/routers/api/notifiers.py

```
@@ -142,6 +144,38 @@ def update_notifier(
+                    status_code=status.HTTP_400_BAD_REQUEST,
+                    detail=f"Invalid email address: {e}",
+                ) from e
+        elif new_type == NotifierType.GENERIC_WEBHOOK:
+            try:
+                # Use TypeAdapter(HttpUrl) for Pydantic V2 compatibility
+                url = TypeAdapter(HttpUrl).validate_python(new_address)
+                if url.scheme != "https":
+                    raise HTTPException(
+                        status_code=status.HTTP_400_BAD_REQUEST,
+                        detail="Webhook URLs must use HTTPS for security",
+                    )
```

Ruff (0.15.0)

[warning] Abstract `raise` to an inner function

(TRY301)